syntax = "proto3";

package persistence;

import "symbols.proto";
import "relocs.proto";

message ProgramLoaderBinary {
    string filePath = 1;
    bytes data = 2;
}

// What program loader the project uses
enum ProgramLoaderType {
    LOADER_UNSPECIFIED = 0;
    LOADER_RVL = 1;
    LOADER_ELF = 2;
}

// Program loader persistence
message ProgramLoader {
    int32 formatVersion = 1;

    ProgramLoaderType loaderType = 2;
    repeated ProgramLoaderBinary binaries = 3;
    SymbolTable symtab = 4;
    RelocTable reloctab = 5;
}